configure.ac

```diff
@@ -111,6 +111,7 @@ AS_CASE([$host],
   [ARCH_DIR=undefined]
 )
 AC_SUBST([ARCH_DIR])
+AM_CONDITIONAL([ARCH_X86], [test "x${ARCH_DIR}" = "xx86"])
```

Comment:
Please remove this line and use ARCH_IS_X86 instead.  Maybe move ARCH_IS_XXX lines right here if you think that is better.